### PR TITLE
MDN CSS: Add redirects for properties which have URLs pointing to other properties

### LINKED ISSUE #545: MDN CSS: -webkit-* properties should have redirects when no standalone page exists
### Description
There are several `-webkit-*` properties that we don't show answers for, because on MDN those properties simply redirect to other pages.

E.g. `-webkit-transform` is just a link to `transform` on the [Webkit Extensions index](https://developer.mozilla.org/en-US/docs/Web/CSS/Webkit_Extensions)

If we detect that the linked page for the property maps to a link we already have, we should create a redirect entry in the `output.txt`.

That way a search for "webkit-transform" will show an Instant Answer for the CSS `transform` property.

## People to notify
@rasikapohankar

## Get Started
- [ ] 1) Claim this issue by commenting below
- [ ] 2) Review our [Contributing Guide](https://github.com/duckduckgo/zeroclickinfo-spice/blob/master/CONTRIBUTING.md)
- [ ] 3) [Set up your development environment](https://docs.duckduckhack.com/welcome/setup-dev-environment.html), and fork this repository
- [ ] 4) Create a Pull Request

## Resources
- Join [DuckDuckHack Slack](https://quackslack.herokuapp.com/) to ask questions
- Join the [DuckDuckHack Forum](https://forum.duckduckhack.com/) to discuss project planning and Instant Answer metrics
- Read the [DuckDuckHack Documentation](https://docs.duckduckhack.com/) for technical help

------
IA Page:  http://duck.co/ia/view/mdn_css 
[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @hchienjo

### PR DESCRIPTION
**Pull request by rasikapohankar**
## Description of new Instant Answer, or changes

Add redirects for `-moz-*` and `-webkit-*` properties with URLs pointing to other properties

## Related Issues and Discussions

Fixes #545

## People to notify

@moollaza @hchienjo 

---

Instant Answer Page: https://duck.co/ia/view/mdn_css
Maintainer: @hchienjo

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: lib/fathead/mdn_css/fetch.pl [hunk 1]
```diff
@@ -32,14 +33,21 @@
...
 
 #see http://mojolicious.org/perldoc/Mojo/IOLoop#recurring
 Mojo::IOLoop->recurring(
     0 => sub {
         for ( $current_active_connections + 1 .. $maximum_active_connections ) {
-            return ( $current_active_connections or Mojo::IOLoop->stop )
-              unless my $url = shift @keyword_urls;
+            my $url = shift @keyword_urls;
+            if(!$url) {
+                DumpFile('redirect_urls.yml', { redirects => {%redirect_map} } );
```
Comment (hchienjo) [inline lib/fathead/mdn_css/fetch.pl]: No, the Dump will happen for every `tick` of the IOLoop, what you'd want to do is collect whatever you want in the loop and then write the `dump`-ing logic at the bottom, after the IOLoop code around [here](https://github.com/rasikapohankar/zeroclickinfo-fathead/blob/879b07a4040376d60ad2d8202af2656207a80250/lib/fathead/mdn_css/fetch.pl#L86). Judging from the output, the solution looks really nice :+1:.

Comment (rasikapohankar) [inline lib/fathead/mdn_css/fetch.pl]: Thanks @hchienjo 👍 I had noticed the dumping was happening more than once but was unable to figure where to put it 😄 I have moved it to the bottom now.

**Comment by hchienjo** [comment]
@rasikapohankar Shouldn't the the data be written to output.txt as redirect entries?

**Comment by rasikapohankar** [comment]
@hchienjo Yes, that is happening in `parse.pl`. Checked if the some property is a redirect to the title in redirect_urls.yml and if so created a redirect from that property to the title.

**Comment by hchienjo** [comment]
@rasikapohankar great. Sorry I hadn't seen that.

**Comment by hchienjo** [comment]
Looks good on codio too. 
codio.

Looks like there is no problem with this.

**Comment by moollaza** [comment]
@hchienjo Deploying to beta shortly..

**Comment by moollaza** [comment]
@rasikapohankar @hchienjo looks great! Live on Beta: 
https://beta.duckduckgo.com/?q=webkit-transform&t=opera&ia=about

File: lib/fathead/mdn_css/redirect_urls.yml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+  '@document': '@-moz-document'
+  Introduction_to_the_CSS_box_model: box_model
+  Using_CSS_animations: Using_CSS_animations
+  Using_CSS_counters: Counters
+  Using_CSS_flexible_boxes: Flexible_boxes
+  Using_CSS_gradients: Using_CSS_gradients
+  Using_CSS_multiple_backgrounds: Using_multiple_backgrounds
+  Using_CSS_transforms: Using_CSS_transforms
+  Using_CSS_transitions: Using_CSS_transitions
+  Using_media_queries: Media_queries
```
Comment (moollaza) [inline lib/fathead/mdn_css/redirect_urls.yml]: Shouldn't the redirect here be "media queries"? i.e. no underscores

File: lib/fathead/mdn_css/redirect_urls.yml [hunk 3]
```diff
@@ -0,0 +1,112 @@
+---
+redirects:
+  -moz-appearance: -webkit-appearance
+  :any: :-moz-any
+  :any-link: :-moz-any-link
+  :fullscreen: :-moz-full-screen
```
Comment (moollaza) [inline lib/fathead/mdn_css/redirect_urls.yml]: Do we also have redirects for the normalized title? i.e. `moz full screen` -> `:fullscreen`

File: lib/fathead/mdn_css/redirect_urls.yml (hunk 2 above)
Comment (hchienjo) [inline lib/fathead/mdn_css/redirect_urls.yml]: Looking into it.

File: lib/fathead/mdn_css/redirect_urls.yml (hunk 3 above)
Comment (moollaza) [inline lib/fathead/mdn_css/redirect_urls.yml]: Sure, a separate PR would be good 👍

**Pull request merged**